src/subscribe-substrate/lib/utils.ts

```diff
@@ -19,3 +24,52 @@ export const getEventMethods = () => {
         return ["all"];
     }
 };
+
+export type InsertData = BlogId | PostId | CommentId;
+
+export function encodeStructId(id: InsertData): string {
+    if (!id) return null;
+
+    return id.toHex().split('x')[1].replace(/(0+)/, '');
+}
+
+export async function insertElasticSearch<T extends IpfsData>(index: string, ipfsHash: string, id: InsertData | AccountId, extData?: object) {
+    const json = await getJsonFromIpfs<T>(ipfsHash);
+    let indexData;
+
+    switch (index) {
+        case process.env.BLOGS_INDEX: {
+            const { name, desc } = json as BlogData;
+            indexData = { name, desc };
+            break;
+        }
+
+        case process.env.POSTS_INDEX: {
+            const { title, body } = json as PostData;
+            indexData = { title, body };
+            break;
+        }
+
+        case process.env.COMMENTS_INDEX: {
+            const { body } = json as CommentData;
+            indexData = { body };
+            console.log('[*]\n' + indexData + '\n[*]');
+            break;
+        }
+
+        case process.env.PROFILES_INDEX: {
+            const { fullname, about } = json as ProfileData;
+            indexData = { ...extData, fullname, about };
+            break;
+        }
+
+        default:
+            break;
+    }
+
+    await searchClient.index({
```

Comment:
Add `if (indexData) {` ... `}` around `await searchClient.index...`.

Because if `default` case happened, then `indexData` will not be defined